nano/node/backlog_population.cpp

```diff
@@ -0,0 +1,90 @@
+#include <nano/node/backlog_population.hpp>
+#include <nano/node/node.hpp>
+
+nano::backlog_population::backlog_population (nano::node & node_a) :
+	node{ node_a }
+{
+}
+
+nano::backlog_population::~backlog_population ()
+{
+	stop ();
+	if (thread.joinable ())
+	{
+		thread.join ();
+	}
+}
+
+void nano::backlog_population::start ()
+{
+	if (!thread.joinable ())
+	{
+		thread = std::thread{ [this] () { run (); } };
+	}
+}
+
+void nano::backlog_population::stop ()
+{
+	nano::unique_lock<nano::mutex> lock{ mutex };
+	stopped = true;
+	notify ();
+}
+
+void nano::backlog_population::trigger ()
+{
+	nano::unique_lock<nano::mutex> lock{ mutex };
+	triggered = true;
+	notify ();
+}
+
+void nano::backlog_population::notify ()
+{
+	condition.notify_all ();
+}
+
+bool nano::backlog_population::predicate () const
+{
+	return triggered;
+}
+
+void nano::backlog_population::run ()
+{
+	nano::thread_role::set (nano::thread_role::name::backlog_population);
+	nano::unique_lock<nano::mutex> lock{ mutex };
+	while (!stopped)
+	{
+		if (predicate () || (node.config.frontiers_confirmation != nano::frontiers_confirmation_mode::disabled))
+		{
+			lock.unlock ();
+			populate_backlog ();
+			lock.lock ();
+			triggered = false;
```

Comment:
This should be moved to before calling  populate_backlog() so that a trigger cannot possibly be missed i the case that a trigger comes in at the end of populate backlog cycle.